### PR TITLE
DTD-441 plan Added payment plan test scenarios

### PR DESCRIPTION
**Pull request by thadeusssewanyana**

### CODE REVIEW

**Review by vivekhmrc — COMMENTED** [review]
Reviewing

File: run_statement_of_liability_api_pega_contract_tests.sh [hunk 1]
```diff
@@ -10,4 +10,4 @@ fi
 
 echo "*** running on $environment for tags '$tags' ***"
 
-sbt -Denvironment="$environment" -Dcucumber.options="--tags '$tags'" clean 'testOnly uk.gov.hmrc.test.api.cucumber.runner.StatementOfLiabilityApiTestRunner'
\ No newline at end of file
+sbt -Denvironment="$environment" -Dcucumber.options="--tags '@smoke'" clean 'testOnly uk.gov.hmrc.test.api.cucumber.runner.StatementOfLiabilityApiPegaTestRunner'
```
Comment (vivekhmrc) [inline run_statement_of_liability_api_pega_contract_tests.sh]: Is _smoke_ needed here or the value needs to be driven from _tags_ ?

File: src/test/scala/uk/gov/hmrc/test/api/conf/TestConfiguration.scala [hunk 2]
```diff
@@ -27,15 +27,14 @@ object TestConfiguration {
...
+      case _       => s"${envConfig.getString(s"services.$service.host")}"
     }
     s"$host${serviceRoute(service)}"
   }
 
-  def zapProxy: Boolean = envConfig.getBoolean("zapProxy")
-
-  def environmentHost: String = envConfig.getString("services.host")
-
+  def environmentHost: String                  = envConfig.getString("services.host")
```
Comment (vivekhmrc) [inline src/test/scala/uk/gov/hmrc/test/api/conf/TestConfiguration.scala:35]: Should this be _s"services.$service.host"_?

File: src/test/scala/uk/gov/hmrc/test/api/utils/BaseRequests.scala [hunk 3]
```diff
@@ -1,43 +1,119 @@
...
+
+    TestConfiguration.env match {
+      case "local" =>
+        createLocalBearerToken(enrolments, userType, utr)
+      case _       =>
+        createQaBearerToken(enrolments.head)
+
+    }
+
+    //  def createBearerToken(
```
Comment (vivekhmrc) [inline src/test/scala/uk/gov/hmrc/test/api/utils/BaseRequests.scala]: Do we need this commented code ?

**Review by vivekhmrc — COMMENTED** [review]
Left a few minor observations @thadeusssewanyana

**Review by vivekhmrc — APPROVED** [review]
LGTM 👍

**Review by rogerball — APPROVED** [review]

**Pull request merged**